### ISSUE TITLE
Onboard Librem

### ISSUE THREAD

**Issue by samcday** [samcday/home-cluster#475]
My old Librem 13 laptop has participated in various previous iterations of the home-cluster (for a while it was *the* "home-cluster"!). It hasn't been in the cluster for quite a few months properly.

I want to re-add it as a non control-plane node. I'll need to rework the Butane/Ignition config a little to accommodate this.

I think #408 is related because while I'm in the area chopping up the config and thinking about the secret handling etc, I might as well finally figure out how to ensure that the cluster can handle (re-)provisioning its own node members.

The Librem 13 in particular is quirky - CoreOS doesn't seem to boot properly. After GRUB it's just a black screen. I first noticed this back in FCOS36/37, and it's still the case today with FCOS40.

Silverblue/Workstation work fine, though. So the plan is to specialize an image from my [workstation-config](https://github.com/samcday/workstation-config) and run CoreOS as a libvirt guest in a Silverblue host.

**Comment by samcday** [comment]
I got something basic working this afternoon by `virt-install`ing FCOS in Silverblue. I don't hate it, but I don't love it either.

I realized during the process that I might just be able to layer/hack on Silverblue base a bit more and run the k8s node directly there?

I've done a bunch of work on the provisioning such that a non-control-plane node really only needs a join token so, in theory this approach should be straightforward. If I can get it to work.

**Comment by samcday** [comment]
Getting a Silverblue host through the basic `kubeadm join` (+ preflight) was straightforward enough. So now the Librem is joined directly to the cluster at the host level. I still need to properly capture the changes in a layered `Dockerfile`, though.

The node isn't fully functional yet, I initially let it join untainted and it adopted a bunch of workloads that started choking.

I should have it all sorted today.

**Comment by samcday** [comment]
I've stabilized the Librem as a cluster member running Silverblue. Once I brought across the (unfortunately a little duplicated) baseline k8s node setup from `node/`, and disabled `firewalld.service`, everything seems pretty gucci.

I deployed descheduler this morning to ensure that workloads are somewhat evenly spread across all the nodes. So the Librem is running its fair share of random bits of and pieces. Alertmanager is clear.

In [40eb7ef](https://github.com/samcday/home-cluster/commit/40eb7ef1855312055d2cd354251dc5dabe3ba006) I tied a bow on the work by automating the Librem daily image update the same way as the main control-plane nodes.,
